lib/dry/web/settings.rb

```diff
@@ -0,0 +1,59 @@
+require "yaml"
+
+module Dry
+  module Web
+    class Settings
+      TypeError = Class.new(StandardError)
+
+      AnyType = Class.new do
+        def self.[](value)
+          value
+        end
+      end
```

Comment:
Ah, this is an interesting point, actually. I was actually building this so it can work without any dependency on dry-types, basically so that we're not forcing it on people, instead just building things that would work _well_ with it if they choose to use it. 

Does that seem ok with you, or do you think we should build in a dry-types dependency here?

An alternative to AnyType here would just be to have nil as the default for the type argument and only run it through the typecasting object if one is supplied. 

> On 16 Jun 2016, at 5:54 PM, Piotr Solnica <notifications@github.com> wrote:
> 
> In lib/dry/web/settings.rb:
> 
> > @@ -0,0 +1,59 @@
> > +require "yaml"
> > +
> > +module Dry
> > +  module Web
> > +    class Settings
> > +      TypeError = Class.new(StandardError)
> > +
> > +      AnyType = Class.new do
> > +        def self.[](value)
> > +          value
> > +        end
> > +      end
> We should probably add Dry::Types::Object for such use-cases, could you add a TODO comment so that we won't forget to update it once this type is in place?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>